src/strands/tools/mcp/mcp_agent_tool.py

```diff
@@ -28,26 +28,29 @@ class MCPAgentTool(AgentTool):
     seamlessly within the agent framework.
     """
 
-    def __init__(self, mcp_tool: MCPTool, mcp_client: "MCPClient") -> None:
+    def __init__(self, mcp_tool: MCPTool, mcp_client: "MCPClient", agent_facing_tool_name: str | None = None) -> None:
         """Initialize a new MCPAgentTool instance.
 
         Args:
             mcp_tool: The MCP tool to adapt
             mcp_client: The MCP server connection to use for tool invocation
+            agent_facing_tool_name: Optional name to use for the agent tool (for disambiguation)
```

Comment:
I'd suggest something less verbose - `name_override` or just `name` with documentation intending the purpose?